tests/testsuite/git_auth.rs

```diff
@@ -277,31 +300,74 @@ fn ssh_something_happens() {
             // "[..]banner exchange: Connection to 127.0.0.1 [..]"
             //   banner exchange: Connection to 127.0.0.1 port 62250: Software caused connection abort
             // But since there is no common meaningful sequence or word, we can only match a small telling sequence of characters.
-            "[..]onnect[..]"
+            format!(
+                "  banner exchange: Connection to {} port {}: Software caused connection abort",
+                addr.ip(),
+                addr.port()
+            )
         } else {
-            "[..]Connection [..] by [..]"
+            "[..]Connection closed by [..]".to_string()
```

Comment:
Could just revert the assertion string of `message` back? By reading the comment I am afraid we'll bump into some error for different targets (musl vs gnu vs MSVC I guess?).

Reply:
Seems not just `message`, it is flaky sometimes with following pattern.

```
thread 'git_auth::ssh_something_happens' panicked at tests/testsuite/git_auth.rs:347:10:

--- Expected
++++ actual:   stderr
   1    1 | [UPDATING] git repository `ssh://127.0.0.1:45997/foo/bar`
        2 + kex_exchange_identification: read: Connection reset by peer
        3 + Connection reset by 127.0.0.1 port 45997
        4 + [WARNING] spurious network error (3 tries remaining): failed to fill whole buffer
   2    5 | [ERROR] failed to get `bar` as a dependency of package `foo v0.0.1 ([ROOT]/foo)`
   3    6 | 
   4    7 | Caused by:
   5    8 |   failed to load source for dependency `bar`
   6    9 | 
          ⋮
  17   20 | 
  18   21 | Caused by:
  19   22 |   An IO error occurred when talking to the server
  20   23 | 
  21   24 | Caused by:
  22      - [..]Connection [..] by [..]
       25 +   ssh: connect to host 127.0.0.1 port 45997: Connection refused
```
instead of current snapshot with last line `  Connection closed by {ip} port {port}`

Could this expected string be below one?
```
[UPDATING] git repository `ssh://127.0.0.1:45997/foo/bar`
...
{message}
...
```

Reply:
> Seems not just message, it is flaky sometimes in local testing with following pattern.

That's the reason I fear for. We shouldn't assert outputs from external tools (ssh, git). 
I am fine with your suggestion. Go ahead.